### PR TITLE
replication manager post job test

### PR DESCRIPTION
**Pull request by rajkumarpatel2602**
* Add pytest for fdmi record post in replication manager

### CODE REVIEW

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py [hunk 1]
```diff
@@ -91,22 +93,19 @@ async def add_job(request):
     """
     job_record = await request.json()
 
-    # Get first key and find if already present
-    job_id = job_record['Object-Name']
-    job = request.app['all_jobs'].get_job(job_id)
+    job_prep = PrepareJob()
```
Comment (kaustubh-d) [inline s3/replication/manager/src/s3replicationmanager/manager_routes.py:96]: FdmiToReplicationJobConv

Comment (kaustubh-d) [reply s3/replication/manager/src/s3replicationmanager/manager_routes.py:96]: Another option PrepareReplicationJob

File: s3/replication/manager/tests/system/test_replication_manager.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        port = str(config['manager'].get('port'))
+
+    # URL for non-secure http endpoint
+    url = 'http://' + host + ':' + port
+
+    return {'url': url, 'fdmi_job': fdmi_job}
+
+
+@pytest.mark.asyncio
+async def test_post_job(test_config):
```
Comment (kaustubh-d) [inline s3/replication/manager/tests/system/test_replication_manager.py:90]: Add valid test input and empty test input test case and corrupt input (like missing fields)

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py [hunk 3]
```diff
@@ -76,22 +76,21 @@ async def add_job(request):
...
-        LOG.debug('Added Job with job_id : {} '.format(job.get_job_id()))
-        LOG.debug(
-            'Added Job : {} '.format(
-                json.dumps(
-                    job,
-                    cls=JobJsonEncoder)))
-        return web.json_response({'job': job.get_dict()}, status=201)
+    job_prep = PrepareReplicationJob()
+
+    fdmi_job = job_prep.prepare_replication_job(job_record)
```
Comment (kaustubh-d) [inline s3/replication/manager/src/s3replicationmanager/manager_routes.py:84]: names look reversed, fdmi_job here is actually converted replication job and job_record is fdmi_record

File: s3/replication/manager/src/s3replicationmanager/prepare_job.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+class PrepareReplicationJob():
+
+    def __init__(self):
+        """Replication manager's job id."""
+
+        # Job id for replication manager's reference
+        self.rm_job_id = str(uuid.uuid4())
+
+    def prepare_replication_job(self, fdmi_record):
```
Comment (kaustubh-d) [inline s3/replication/manager/src/s3replicationmanager/prepare_job.py:41]: can be named 
`def from_fdmi(fdmi_record)`
usage 
`replication_job = PrepareReplicationJob.from_fdmi(fdmi_record)`

File: s3/replication/manager/tests/system/test_replication_manager.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        port = str(config['manager'].get('port'))
+
+    # URL for non-secure http endpoint
+    url = 'http://' + host + ':' + port
+
+    return {'url': url}
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('test_input_param', ['valid_job', 'empty_job'])
```
Comment (kaustubh-d) [inline s3/replication/manager/tests/system/test_replication_manager.py:97]: Can this be simply written as 
```
@pytest.mark.parametrize('input_param', [fdmi_job, {}])
async def test_post_job(test_config, input_param):
...
```

File: s3/replication/manager/tests/system/test_replication_manager.py [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+async def test_post_job(test_config, test_input_param):
+    """Pytest for post job test."""
+    input_test = {'valid_job': fdmi_job_record, 'empty_job': {}}
```
Comment (kaustubh-d) [inline s3/replication/manager/tests/system/test_replication_manager.py:100]: If above suggestion works, this wont be required

File: s3/replication/manager/src/s3replicationmanager/manager_routes.py (hunk 3 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/src/s3replicationmanager/manager_routes.py:84]: done

File: s3/replication/manager/tests/system/test_replication_manager.py (hunk 2 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/tests/system/test_replication_manager.py]: done

File: s3/replication/manager/tests/system/test_replication_manager.py (hunk 5 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/tests/system/test_replication_manager.py:97]: There's  limitation with parametrize fixture, that one can not pass variable value in it.
Hence I have used the list internally with 'valid_job' and 'empty_job' keys.

File: s3/replication/manager/tests/system/test_replication_manager.py (hunk 6 above)
Comment (rajkumarpatel2602) [reply s3/replication/manager/tests/system/test_replication_manager.py:100]: Getting ahead with existing change due to limitation.

**Review by kaustubh-d — APPROVED** [review]
